src/control/server/ctl_storage.go

```diff
@@ -34,11 +34,28 @@ import (
 	"github.com/daos-stack/daos/src/control/server/storage/scm"
 )
 
+type (
+	scmProvider interface {
```

Comment:
shouldn't this consist of 2 embedded interface types, scmBackend (satisfied by cmdRunner for ndctl and ipmct bindings) & sysBackend then drop in replacements can be provided for each with clean identification for what is covering which part of the interface definition

Reply:
I don't think so. At this level I don't think we should be exposing the implementation details. The intent behind scm.Provider was to expose a unified set of functionality without making the user of that functionality be concerned with the details of how it is provided. We may change the implementation to depend more on the bindings, for example, and ideally users of scm.Provider won't need to know about those changes.

As I noted in our skype chat, I think that the tests at this level should be simplified and primarily concerned about validating that the correct RPC responses are generated for various inputs, which means that the mock at this level can be pretty basic.

Reply:
ok, so keep that detail and testing of in server/scm

Reply:
Right, exactly. It allows us to make the tests at this level simpler because they don't need to exercise all of the scenarios from out here.